espnet/nets/pytorch_backend/e2e_asr_transducer.py

```diff
@@ -0,0 +1,414 @@
+                seq_true_text = "".join(seq_true).replace(self.recog_args.space, ' ')
+
+                hyp_words = seq_hat_text.split()
+                ref_words = seq_true_text.split()
+                word_eds.append(editdistance.eval(hyp_words, ref_words))
+                word_ref_lens.append(len(ref_words))
+
+                hyp_chars = seq_hat_text.replace(' ', '')
+                ref_chars = seq_true_text.replace(' ', '')
+                char_eds.append(editdistance.eval(hyp_chars, ref_chars))
+                char_ref_lens.append(len(ref_chars))
+            wer = 0.0 if not self.report_wer else float(sum(word_eds)) / sum(word_ref_lens)
+            cer = 0.0 if not self.report_cer else float(sum(char_eds)) / sum(char_ref_lens)
+
+        self.loss = loss
+        loss_data = float(self.loss)
+
+        if not math.isnan(loss_data):
+            self.reporter.report(loss_data, cer, wer)
+        else:
+            logging.warning('loss (=%f) is not correct', loss_data)
+
+        return self.loss
+
+    def recognize(self, x, recog_args, char_list, rnnlm=None):
+        """E2E recognize
+
+        Args:
+            x (ndarray): input acoustic feature (T, D)
+            recog_args (namespace): argument Namespace containing options
+            char_list (list): list of characters
+            rnnlm (torch.nn.Module): language model module
+
+        Returns:
+           y (list): n-best decoding results
+        """
+
+        prev = self.training
+        self.eval()
+        ilens = [x.shape[0]]
+
+        # subsample frame
+        x = x[::self.subsample[0], :]
+        h = to_device(self, to_torch_tensor(x).float())
+        # make a utt list (1) to use the same interface for encoder
+        hs = h.contiguous().unsqueeze(0)
+
+        # 0. Frontend
+        if self.frontend is not None:
+            enhanced, hlens, mask = self.frontend(hs, ilens)
+            hs, hlens = self.feature_transform(enhanced, hlens)
+        else:
+            hs, hlens = hs, ilens
+
+        # 1. Encoder
+        h, _, _ = self.enc(hs, hlens)
+
+        # 2. Decoder (pred+joint or att-dec+joint)
+        if recog_args.search_type == 'greedy':
+            y = self.dec.recognize(h[0], recog_args)
+        else:
+            y = self.dec.recognize_beam(h[0], recog_args, rnnlm)
+
+        if prev:
+            self.train()
+
+        return y
+
+    def recognize_batch(self, xs, recog_args, char_list, rnnlm=None):
+        """E2E recognize with batch
+
+        Args:
+            xs (ndarray): list of input acoustic feature arrays [(T_1, D), (T_2, D), ...]
+            recog_args (namespace): argument Namespace containing options
+            char_list (list): list of characters
+            rnnlm (torch.nn.Module): language model module
+
+        Returns:
+           y (list): n-best decoding results
+        """
+
+        logging.error("Batch decoding for transducer is not supported yet.")
+        raise NotImplementedError
```

Comment:
You can simply remove this method because this method is dead code

Reply:
I should not raise an error if batch decoding is called?

Reply:
I see. Maybe we can move these docstring and `NotImplementedError` to `ASRInterface.recognize_batch` instead of here.

Reply:
I removed the implementation as suggested for now. I can do another PR to do the changes in asr/asr.py when we are set on the approach.